### PR TITLE
BuildService 'nocache.length() == 0' can be replaced with 'nocache.isEmpty()' #3198 Fix

### LINKED ISSUE #3198: BuildService 'nocache.length() == 0' can be replaced with 'nocache.isEmpty()'
### Description

The call to `String.length` can be replaced with `isEmpty` in [BuildService](https://github.com/eclipse-jkube/jkube/blob/9dee5b047481246be704a0408f71fc917fc0ac6c/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/BuildService.java).

The following line:
https://github.com/eclipse-jkube/jkube/blob/9dee5b047481246be704a0408f71fc917fc0ac6c/jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/BuildService.java#L219

should be changed to:

```java
            return nocache.isEmpty() || Boolean.parseBoolean(nocache);
```

### :red_circle: Before you start :red_circle:

:point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down: :point_down:

Make sure you read the [contributing guide](https://www.eclipse.org/jkube/contributing) first.

Pay special attention to the ECA agreement section and the requirement to sign-off your commit.

This is a [first-timers-only](https://eclipse.dev/jkube/contributing/#first-timers-only-issues) issue. These issues are especially created for contributors who are contributing to the project for the **first time**.

If you have already contributed to this project in the past, please don't ask to be assigned one of these issues and look for [good-first-issues](https://github.com/eclipse-jkube/jkube/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22good+first+issue%22) or [help-wanted](https://github.com/eclipse-jkube/jkube/issues?q=is%3Aissue+is%3Aopen+sort%3Aupdated-desc+label%3A%22help+wanted%22+) issues instead in order to continue you contributor journey in the project.

### PR DESCRIPTION
**Pull request by Aknummoalr**
## Description

Fixes #3198

## Type of change

 - [x] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [ ] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] I have read the [contributing guidelines](https://www.eclipse.dev/jkube/contributing)
 - [ ] I signed-off my commit with a user that has signed the [Eclipse Contributor Agreement](https://www.eclipse.org/legal/ECA.php)
 - [x] My code follows the style guidelines of this project
 - [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
 - [ ] I use [conventional commits](https://www.conventionalcommits.org/) in my commit messages
 - [x] I have performed a self-review of my code
 - [ ] I Added [CHANGELOG](../CHANGELOG.md) entry
 - [ ] I have updated the [documentation](../kubernetes-maven-plugin/doc) accordingly
 - [ ] No new bugs, code smells, etc. in [SonarCloud](https://sonarcloud.io/dashboard?id=jkubeio_jkube) report
 - [ ] I have added tests that prove my fix is effective or that my feature works
 - [ ] New and existing unit tests pass locally with my changes
 - [ ] I tested my code in Kubernetes
 - [ ] I tested my code in OpenShift

### CODE REVIEW

**Comment by manusa** [comment]
### Eclipse JKube [CI Report](https://github.com/eclipse-jkube/ci)

Started new GH workflow run for https://github.com/eclipse-jkube/jkube/pull/3434 (_2024-10-08T15:04:35Z_)

:gear: [JKube E2E Tests (11238295569)](https://github.com/eclipse-jkube/ci/actions/runs/11238295569) :hourglass_flowing_sand:
- :hourglass_flowing_sand: [Bootstrap CI test run (#3434)](https://github.com/eclipse-jkube/ci/actions/runs/11238295569/job/31242717411)

**Comment by rohanKanojia** [comment]
@Aknummoalr Welcome, Thanks a lot for contributing to Eclipse JKube :+1:,

In order to contribute to Eclipse, you would need to create an [Eclipse Account](https://accounts.eclipse.org/) and Sign [Eclipse Contributor Agreement](https://accounts.eclipse.org/user/login?destination=user/eca). Once you've created an eclipse account you would need to update your commit message with a `Signed-off-by:` footer. You can do it like this:
```
# (Optional): Configure Git User and Email
git config user.name "Your Name"
git config user.email "your_email@example.com"

# Amend your previous commit with signed off footer
git commit --amend --signoff

# push changes again
git push origin yourbranchname -f
```

File: jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/BuildService.java~ [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.build.service.docker;
```
Comment (rohanKanojia) [inline jkube-kit/build/service/docker/src/main/java/org/eclipse/jkube/kit/build/service/docker/BuildService.java~:14]: Please remove this file, it's a probably a backup file you accidentally added to your git index.

**Comment by Aknummoalr** [comment]
I removed the files requested by @rohanKanojia

File: quickstarts/README.md [hunk 2]
```diff
@@ -15,4 +15,4 @@ To override the JKube release you can pass the `jkube.version` property when run
 e.g.
 ```shell script
 $ mvn -Pkubernetes clean install k8s:build -Djkube.version=1.33.7
-```
\ No newline at end of file
+```
```
Comment (rohanKanojia) [inline quickstarts/README.md:18]: Could you please revert changes done to this file? It's not clear in diff what changes have been made.
You should only keep files relevant to the issue you've linked in the pull request.

**Review by rohanKanojia — APPROVED** [review]

**Review by manusa — APPROVED** [review]
LGTM, thx!

**Pull request merged**